esmvalcore/preprocessor/_time.py

```diff
@@ -327,7 +327,28 @@ def annual_statistics(cube, operator='mean'):
 
     if not cube.coords('year'):
         iris.coord_categorisation.add_year(cube, 'time')
-    return cube.aggregated_by('year', operator)
+    years = sorted(list(set(cube.coord("year").points)))
```

Comment:
not sure, I'd rather not since that looks to me like it's a bit clunky - I'd rather not repeat all the memory profiling tests for a rewrite of the same functionality :grin:

Reply:
I was just curious, as you write lots of code for something that can be a one liner